examples/ex18.cpp

```diff
@@ -0,0 +1,298 @@
+   //    opened with GLvis with the -gc option.
+   Array<int> offsets(num_equation + 1);
+   for (int k = 0; k <= num_equation; k++) { offsets[k] = k * vfes.GetNDofs(); }
+   BlockVector u_block(offsets);
+
+   // Density grid function for visualization.
+   GridFunction mom(&dfes, u_block.GetData() + offsets[1]);
+
+   // Initialize the block vector state: define a vector finite element space
+   // for all equations and initialize together.
+   VectorFunctionCoefficient u0(num_equation, u0_function);
+   GridFunction sol(&vfes, u_block.GetData());
+   sol.ProjectCoefficient(u0);
+
+   // Output the initial solution.
+   {
+      ofstream mesh_ofs("vortex.mesh");
+      mesh_ofs.precision(precision);
+      mesh_ofs << mesh;
+
+      for (int k = 0; k < num_equation; k++)
+      {
+         GridFunction uk(&fes, u_block.GetBlock(k));
+         ostringstream sol_name;
+         sol_name << "vortex-" << k << "-init.gf";
+         ofstream sol_ofs(sol_name.str().c_str());
+         sol_ofs.precision(precision);
+         sol_ofs << uk;
+      }
+   }
+
+   // 7. Set up the nonlinear form corresponding to the DG discretization of the
+   //    flux divergence, and assemble the corresponding mass matrix.
+   MixedBilinearForm Aflux(&dfes, &fes);
+   Aflux.AddDomainIntegrator(new DomainIntegrator(dim));
+   Aflux.Assemble();
+
+   NonlinearForm A(&vfes);
+   RiemannSolver rsolver;
+   A.AddInteriorFaceIntegrator(new FaceIntegrator(rsolver, dim));
+
+   // 8. Define the time-dependent evolution operator describing the ODE
+   //    right-hand side, and perform time-integration (looping over the time
+   //    iterations, ti, with a time-step dt).
+   FE_Evolution euler(vfes, A, Aflux.SpMat());
+
+   // Visualize the density
+   socketstream sout;
+   if (visualization)
+   {
+      char vishost[] = "localhost";
+      int  visport   = 19916;
+
+      sout.open(vishost, visport);
+      if (!sout)
+      {
+         cout << "Unable to connect to GLVis server at "
+              << vishost << ':' << visport << endl;
+         visualization = false;
+         cout << "GLVis visualization disabled.\n";
+      }
+      else
+      {
+         sout.precision(precision);
+         sout << "solution\n" << mesh << mom;
+         sout << "pause\n";
+         sout << flush;
+         cout << "GLVis visualization paused."
+              << " Press space (in the GLVis window) to resume it.\n";
+      }
+   }
+
+   // Determine the minimum element size.
+   double hmin;
+   if (cfl > 0)
+   {
+      hmin = mesh.GetElementSize(0, 1);
+      for (int i = 1; i < mesh.GetNE(); i++)
+      {
+         hmin = min(mesh.GetElementSize(i, 1), hmin);
+      }
+   }
+
+   // Start the timer.
+   tic_toc.Clear();
+   tic_toc.Start();
+
+   double t = 0.0;
+   euler.SetTime(t);
+   ode_solver->Init(euler);
+
+   if (cfl > 0)
+   {
+      // Find a safe dt, using a temporary vector.
+      Vector z(A.Width());
+      max_char_speed = 0.;
+      A.Mult(sol, z);
```

Comment:
I think you should remind that 'max_char_speed' is modified by integrators.

Reply:
Thanks! Addressed in 34009d2